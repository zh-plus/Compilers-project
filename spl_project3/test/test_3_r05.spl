int foo(int i, int j, int k){
    return i - j + k*2;
}

int main(){
    int a = 110;
    int b = 97;
    int c = 3;
    c = foo(a,b,c);
    write(c);
    return 0;
}
