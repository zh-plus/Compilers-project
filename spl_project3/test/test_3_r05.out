FUNCTION foo :
PARAM t0
PARAM t1
PARAM t2
t3 := t0 - t1
t4 := t2 * #2
t5 := t3 + t4
RETURN t5
FUNCTION main :
t6 := #110
t7 := #97
t8 := #3
ARG t8
ARG t7
ARG t6
t9 := CALL foo
t8 := t9
WRITE t8
RETURN #0
